runconfig/opts/parse.go

```diff
@@ -351,6 +389,62 @@ func Parse(cmd *flag.FlagSet, args []string) (*container.Config, *container.Host
 		return nil, nil, nil, cmd, err
 	}
 
+	// Healthcheck
+	var healthConfig *container.HealthConfig
+	haveHealthSettings := *flHealthCmd != "" ||
+		*flHealthInterval != noDuration ||
+		*flHealthTimeout != noDuration ||
+		*flHealthGrace != noDuration ||
+		*flHealthRetries != 0 ||
+		flHealthExit.isSet
+	if *flNoHealthcheck {
+		if haveHealthSettings {
+			return nil, nil, nil, cmd, fmt.Errorf("--no-healthcheck conflicts with --health-* options")
+		}
+		test := strslice.StrSlice{"NONE"}
+		healthConfig = &container.HealthConfig{Test: test}
+	} else if haveHealthSettings {
+		var exitOnUnhealthy *bool
+		var probe strslice.StrSlice
+		if *flHealthCmd != "" {
+			var args []string
+			if runtime.GOOS != "windows" {
+				args = []string{"CMD", "/bin/sh", "-c", *flHealthCmd}
+			} else {
+				args = []string{"CMD", "cmd", "/S", "/C", *flHealthCmd}
```

Comment:
This isn't right. Which shell to use should be decided by the daemon, not the client. But I'm not sure what to do here. The other commands in `docker run` (`--entrypoint` and the `run` command) don't use the shell at all. The easiest thing would be to make it work like entrypoint, but that might be inconvenient.

The other option is the change the format to allow e.g. `{"CMD-SHELL", "check.sh", "-q"}` to be sent over the wire, and have the daemon convert to the appropriate shell. Any preference?